game/debriefing.py

```diff
@@ -109,16 +109,43 @@ class StateData:
     base_capture_events: List[str]
 
     @classmethod
-    def from_json(cls, data: Dict[str, Any]) -> StateData:
+    def from_json(cls, data: Dict[str, Any], unit_map: UnitMap) -> StateData:
+        def clean_unit_list(unit_list: List[Any]) -> List[str]:
+            # Cleans list of units in state.json by
+            # - Removing duplicates. Airfields emit a new "dead" event every time a bomb
+            #   is dropped on them when they've already dead.
+            # - Normalise dead map objects (which are ints) to strings. The unit map
+            #   only stores strings
+            units = set()
+            for unit in unit_list:
+                units.add(str(unit))
+            return list(units)
+
+        killed_aircraft = []
+        killed_ground_units = []
+
+        # process killed units from S_EVENT_UNIT_LOST & S_EVENT_KILL
+        killed_units = clean_unit_list(data["unit_lost_events"] + data["kill_events"])
+        for unit in killed_units:  # organize killed units into aircraft vs ground
+            if unit_map.flight(unit) is not None:
+                killed_aircraft.append(unit)
+            else:
+                killed_ground_units.append(unit)
+
+        # process killed units from S_EVENT_CRASH
```

Comment:
Is there any reason to continue listening for this?

Reply:
Yes, we definitely need to continue listening to these events to handle scenarios where either the player or AI flies into the ground without being killed e.g. CFIT.